userspace/libsinsp/test/public_sinsp_API/events_set.cpp

```diff
@@ -125,19 +135,83 @@ TEST(events_set, names_to_event_set)
 	auto event_names = std::unordered_set<std::string>{"openat2","execveat"};
 
 	auto event_codes = libsinsp::events::names_to_event_set(event_names);
-	ASSERT_TRUE(event_codes.contains(PPME_SYSCALL_OPENAT2_E));
-	ASSERT_TRUE(event_codes.contains(PPME_SYSCALL_OPENAT2_X));
-	ASSERT_TRUE(event_codes.contains(PPME_SYSCALL_EXECVEAT_E));
-	ASSERT_TRUE(event_codes.contains(PPME_SYSCALL_EXECVEAT_X));
-	ASSERT_FALSE(event_codes.contains(PPME_GENERIC_E));
-	ASSERT_FALSE(event_codes.contains(PPME_GENERIC_X));
+	static libsinsp::events::set<ppm_event_code> event_codes_truth = {PPME_SYSCALL_OPENAT2_E, PPME_SYSCALL_OPENAT2_X,
+	PPME_SYSCALL_EXECVEAT_E, PPME_SYSCALL_EXECVEAT_X};
+	ASSERT_PPM_EVENT_CODES_EQ(event_codes, event_codes_truth);
+	ASSERT_TRUE(event_codes.equals(event_codes_truth));
 	ASSERT_EQ(event_codes.size(), 4); // enter/exit events for each event name
 
 	// Now insert a syscall bound to a generic event
 	event_names.insert("syncfs");
 	event_codes = libsinsp::events::names_to_event_set(event_names);
-	ASSERT_TRUE(event_codes.contains(PPME_GENERIC_E));
-	ASSERT_TRUE(event_codes.contains(PPME_GENERIC_X));
+	event_codes_truth = {PPME_SYSCALL_OPENAT2_E, PPME_SYSCALL_OPENAT2_X,
+	PPME_SYSCALL_EXECVEAT_E, PPME_SYSCALL_EXECVEAT_X, PPME_GENERIC_E, PPME_GENERIC_X};
+	ASSERT_PPM_EVENT_CODES_EQ(event_codes, event_codes_truth);
+	ASSERT_TRUE(event_codes.equals(event_codes_truth));
 	ASSERT_EQ(event_codes.size(), 6); // enter/exit events for each event name
+}
+
+TEST(events_set, event_set_to_names)
+{
+	static std::set<std::string> orderd_events_names_matching_set = {"kill", "dup"};
+	static libsinsp::events::set<ppm_event_code> events_set = {PPME_SYSCALL_KILL_E, PPME_SYSCALL_KILL_X,
+	PPME_SYSCALL_DUP_1_E, PPME_SYSCALL_DUP_1_X};
+	// TODO fix test for PPME_GENERIC_E, PPME_GENERIC_X after fixing method in another PR
```

Comment:
Not clear to which test you are referring to :thinking:

Reply:
clarified comment.